llvm/test/Analysis/CostModel/AArch64/div.ll

```diff
@@ -7,21 +7,21 @@ define i32 @sdiv() {
 ; CHECK-LABEL: 'sdiv'
 ; CHECK-NEXT:  Cost Model: Found an estimated cost of 10 for instruction: %I128 = sdiv i128 undef, undef
 ; CHECK-NEXT:  Cost Model: Found an estimated cost of 1 for instruction: %I64 = sdiv i64 undef, undef
-; CHECK-NEXT:  Cost Model: Found an estimated cost of 28 for instruction: %V2i64 = sdiv <2 x i64> undef, undef
```

Comment:
It feels like this patch is doing two things at once:

1. Changing the costs of divides when not passing in the new Scalars parameter,
2. Changing the costs of divides when passing in a non-null Scalars.

It's probably worth splitting the changes up into two different PRs so we can see review them independently. It also allows us to monitor the performance impact of each change on it's own.

Reply:
changes in `llvm/test/Analysis/CostModel/AArch64/div.ll` are not completely orthogonal to my patch (or (1) is not completely orthogonal to (2) ).


[This](https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp#L3479C1-L3486C10) piece of code needs change for following reasons:
1. Now that I have extra information about the scalars and information about the scalar type can be passed  i.e. instead of passing Op1Info and Op2Info, I need to pass information about the scalar. I forgot to update this. Will update the patch.
2. The extract cost, which contributes to the entire cost, is considered once instead of twice [here](https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp#L3484) which is not correct I think. 

In short, updating this small piece of code would cause changes in `llvm/test/Analysis/CostModel/AArch64/div.ll`

The perf results, as mentioned in the description, are neutral.

Reply:
@david-arm segregated two things as you mentioned.